### PR TITLE
view: change the view routes

### PR DESCRIPTION
**Pull request by jma**
* Adds document and project detailed view in the invenio-records-ui
  configuration.
* Adds permissions to the detailed, preview and file document views to
  limit the access to the public documents.
* Adds front page route and the global jinja filters to the main
  extension application.
* Adds a new werkzeug converter to limit the view matched routes to the
  existing organisations.
* Use `autoflake` to detect unused import packages.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: sonar/route_converters.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+"""Werkzeug Route Converters."""
+
+from flask import current_app, g
+from werkzeug.routing import BaseConverter, ValidationError
+
+from .modules.organisations.api import OrganisationRecord
+
+
+class OrganisationCodeConverter(BaseConverter):
+    """Werkzeug Organistaion code converter."""
```
Comment (sebastiendeleze) [inline sonar/route_converters.py:27]: Typo `Organistaion`

File: sonar/ext.py [hunk 2]
```diff
@@ -105,6 +107,20 @@ def init_config(self, app):
...
+
+        @app.route('/<org_code:view>')
+        def index(view):
+            """Homepage."""
+            return render_template('sonar/frontpage.html')
+
+        @app.template_filter()
+        def nl2br(string):
+            r"""Replace \n to <br>."""
+            return string.replace('\n', '<br>')
```
Comment (sebastiendeleze) [inline sonar/ext.py:122]: Why define the route here and not in the main `views.py`?

File: tests/ui/documents/test_documents_views.py [hunk 3]
```diff
@@ -130,10 +125,10 @@ def test_create_publication_statement(document):
                       'Impr. Coustaud'
 
 
-def test_nl2br():
+def test_nl2br(app):
     """Test nl2br conversion."""
-    text = 'Multiline text\nMultiline text'
-    assert views.nl2br(text) == 'Multiline text<br>Multiline text'
+    assert render_template_string("{{ 'Multiline text\nMultiline text' | nl2br | safe}}") ==\
+        'Multiline text<br>Multiline text'
```
[the remark this answers is not in the document]
Comment (jma) [reply tests/ui/documents/test_documents_views.py:131]: This test the filter as a filter and not as a simple function.

File: sonar/ext.py (hunk 2 above)
Comment (jma) [reply sonar/ext.py:122]: because `nl2br` si a global filter and not related to the document view only.

**Review by sebastiendeleze — APPROVED** [review]

**Pull request merged**